authentication/serializers.py

```diff
@@ -0,0 +1,106 @@
+from django.contrib.auth import authenticate, get_user_model
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from djoser.serializers import UserCreatePasswordRetypeSerializer, UserSerializer, TokenCreateSerializer
+from rest_framework import serializers
+
+from validation.validate_password import validate_password_long, validate_password_include_symbols
+
+
+User = get_user_model()
+
+
+class UserRegistrationSerializer(UserCreatePasswordRetypeSerializer):
+    person_email = serializers.EmailField(write_only=True)
+    password = serializers.CharField(
+        style={"input_type": "password"}, write_only=True)
+
+    comp_registered = serializers.BooleanField()
+    comp_is_startup = serializers.BooleanField()
+
+    class Meta(UserCreatePasswordRetypeSerializer.Meta):
+        model = User
+        fields = (
+            "person_email",
+            "password",
+            "person_name",
+            "person_surname",
+            "comp_name",
+            "comp_registered",
+            "comp_is_startup",
+        )
+
+    def validate(self, attrs):
+        self.fields.pop("re_password", None)
+        re_password = attrs.pop("re_password")
+        password = attrs.get("password")
+        try:
+            validate_password_long(password)
+            validate_password_include_symbols(password)
+        except ValidationError as error:
+            print(error)
+            raise serializers.ValidationError({"password": error.message})
+        if attrs["password"] == re_password:
+            return attrs
+        else:
+            raise serializers.ValidationError(
+                {"password": "Passwords don't match."})
+
+    def create(self, validated_data: dict):
+        if User.objects.filter(person_email=validated_data["person_email"]):
+            raise serializers.ValidationError(
+                {"error": "Email is already registered"})
+        comp_registered = validated_data.get("comp_registered")
+        comp_is_startup = validated_data.get("comp_is_startup")
+        if not comp_registered and not comp_is_startup:
+            raise serializers.ValidationError({"error":
+                                               "Please choose who you represent"})
+        if comp_registered and comp_is_startup:
+            raise serializers.ValidationError({"error":
+                                               "Please choose either registered or startup"})
```

Comment:
You have implemented some additional checks related to comp_registered and comp_is_startup. This is good, but comparing these values using if-elif-else instead of sequential conditions might be more efficient.

Reply:
if-if statement replaced with if-elif